src/engraving/playback/playbackmodel.cpp

```diff
@@ -819,7 +819,7 @@ void PlaybackModel::removeTrackEvents(const InstrumentTrackId& trackId, const mu
 
     auto upperBound = trackPlaybackData.originEvents.upper_bound(timestampTo);
```

Comment:
often timestampTo is a very low value (lowest possible value represented by time_t), and in this case the upperBound iterator can't be reached by erasing elements starting from the lowerBound iterator.

Reply:
We'll need @RomanPudashkin's review on this. I'm not sure if this is the best fix; alternative/additional things to do might be checking `timestampFrom < timestampTo` and/or `lowerBound < upperBound`. Depends on how this method is supposed to handle cases where those conditions aren't met.
It might be good to find out where those very low values for timestampTo come from, and whether those are a sign of a deeper problem.

Reply:
I know exactly the value comes from, line 726 of playbackmodel.cpp `timestamp_t removeEventsTo = -std::numeric_limits<timestamp_t>::max();`. But even if you change that to 0 the same occurs. To me this is a required sensible safety check, because it's obviously not the case you can be certain that the upper bound iterator can reached by repeatedly erasing the element that the lower bound iterator (the definition of those functions makes no such guarantee.  In fact, in a list with elements in top-to-bottom sort order, the upper_bound() iterator will never point to an element after the one pointed to by the lower_bound() iterator.

Reply:
BTW, calling vector::erase() can *change* the value of vector::end(), so it is absolutely necessary to re-evaluate it inside a loop where you're erasing.